### PR TITLE
Openshift: Deploy oauth

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

This adds the Oauth deployment to openshift, which is a prerequisite for #4171. Its state in this PR is that it doesn't crashloop, which doesn't nececarily mean it will work. Since the PR is huge already, I want to get this in and do further fixups in follow-ups.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

/assign @xrstf

### CODE REVIEW

File: api/pkg/controller/openshift/resources/oauth.go [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+			if cm.Data == nil {
+				cm.Data = map[string]string{}
+			}
+			cm.Data["config.yaml"] = oauthConfig.String()
+			return cm, nil
+		}
+	}
+}
+
+// OauthServiceCreator returns the function to reconcile the external OpenVPN service
```
Comment (xrstf) [inline api/pkg/controller/openshift/resources/oauth.go]: OpenVPN service?

File: api/pkg/controller/openshift/resources/oauth.go [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+func OauthServiceCreator(exposeStrategy corev1.ServiceType) reconciling.NamedServiceCreatorGetter {
+	return func() (string, reconciling.ServiceCreator) {
+		return OauthName, func(se *corev1.Service) (*corev1.Service, error) {
+			se.Labels = resources.BaseAppLabel(name, nil)
+
+			if se.Annotations == nil {
+				se.Annotations = map[string]string{}
+			}
+			if exposeStrategy == corev1.ServiceTypeNodePort {
+				se.Annotations["nodeport-proxy.k8s.io/expose"] = "true"
```
Comment (xrstf) [inline api/pkg/controller/openshift/resources/oauth.go:187]: Is there any reason we have a constant for the other annotation but not for this?

Comment (alvaroaleman) [reply api/pkg/controller/openshift/resources/oauth.go:187]: No, but I'd like to fix that up in a separate PR

**Comment by alvaroaleman** [comment]
@xrstf updated, PTAL

**Pull request merged**